pkg/analysis/namingconventions/initializer.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package namingconventions
+
+import (
+	"fmt"
+	"regexp"
+
+	"golang.org/x/tools/go/analysis"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"sigs.k8s.io/kube-api-linter/pkg/analysis/initializer"
+	"sigs.k8s.io/kube-api-linter/pkg/analysis/registry"
+)
+
+func init() {
+	registry.DefaultRegistry().RegisterLinter(Initializer())
+}
+
+// Initializer returns the AnalyzerInitializer for this
+// Analyzer so that it can be added to the registry.
+func Initializer() initializer.AnalyzerInitializer {
+	return initializer.NewConfigurableInitializer(
+		name,
+		initAnalyzer,
+		false,
+		validateConfig,
+	)
+}
+
+func initAnalyzer(cfg *Config) (*analysis.Analyzer, error) {
+	return newAnalyzer(cfg), nil
+}
+
+// validateConfig implements validation of the conditions linter config.
+func validateConfig(cfg *Config, fldPath *field.Path) field.ErrorList {
+	if cfg == nil {
+		return field.ErrorList{field.Required(fldPath, "configuration is required for the namingconventions linter when it is enabled")}
+	}
+
+	if len(cfg.Conventions) == 0 {
+		return field.ErrorList{field.Required(fldPath.Child("conventions"), "at least one naming convention must be specified when the namingconventions linter is enabled")}
+	}
+
+	return validateConventions(fldPath.Child("conventions"), cfg.Conventions...)
+}
+
+func validateConventions(fldPath *field.Path, conventions ...Convention) field.ErrorList {
+	fieldErrs := field.ErrorList{}
+
+	seenConventions := sets.New[string]()
+
+	for i, convention := range conventions {
+		if seenConventions.Has(convention.Name) {
+			fieldErrs = append(fieldErrs, field.Duplicate(fldPath.Index(i).Child("name"), convention.Name))
+			continue
+		}
+
+		fieldErrs = append(fieldErrs, validateConvention(fldPath.Index(i), convention)...)
+		seenConventions.Insert(convention.Name)
+	}
+
+	return fieldErrs
+}
+
+// The cyclop linter says this has cyclomatic complexity of 15.
```

Comment:
Still true? Where's the `nolint`?

Reply:
Ah, I broke this down into multiple functions so no this is no longer true.